### ISSUE TITLE
Synchronization issue with owncloud desktop client.

### ISSUE THREAD

**Issue by zubanst** [owncloud/documents#528]
Owncloud 8.0.5, Documents 0.9.0, owncloud desktop client 1.8.4.
Imagine a document shared by user1 and user 2 replicated on PC1 and PC2 respectively by the owncloud desktop applications..
User1 makes a change in a word document. Document synchronizes on the server, then on the PC2 of user2. User2 is able to see on PC2 the changes made by user1. User2 connects on the web interface and looks the content of the file from Files application. User2 is able to see the changes made by user1. User2 opens the document with Documents from the web interface. Changes made by user1 are not there. So there is a mismatch between the changes made by a user on a shared document through the desktop client and what a user the documents is shared with, will see on a web interface with Documents application. The behaviour is the same with non shared documents. There is a mismatch between the content of a document updated through the desktop client and what the user see on the web interface with Documents application.

**Comment by interele** [comment]
OwnCloud 8.1.1.3 Documents 0.10.1 Client 1.8.4
Similar if not the same problem - simplified 
Create a document in LibreOffice
Save it to a directory that is sync'd 
Go to browser
Go to Documents 
Edit the document and close
Go to Files 
The document there is the original one not the edited one
Consequently the client never gets the edited version

Then for seemingly random users and documents it works

**Comment by interele** [comment]
Does this help ?

{"reqId":"UoikQBZ2WTXF5VGLN358","remoteAddr":"10.xxx.xxx.xxx","app":"documents","message":"Saving failed. Reason:0 does not belong to session bj44q8qwg725stdp85akc1kbyhowd4","level":2,"time":"2015-08-18T12:51:31+00:00"}

**Comment by VicDeo** [comment]
user has a memberId `0`, which obviously a wrong memberId.
Can you post headers that are sent to the server  while saving a document, I'm interested in  `HTTP_WEBODF_MEMBER_ID` header value?

**Comment by interele** [comment]
How do I do that ? It's set up as https only

**Comment by interele** [comment]
webodf-session-id: 0uweob9qqvllqb7h7okcyxc62fjmds
webodf-member-id: 0
webodf-session-revision: 675

**Comment by VicDeo** [comment]
@interele can you also open this document, made some changes and post here data that is passed with request to `../apps/documents/ajax/otpoll.php`
```
args: {es_id: "xxanna4crahd98q5gec8kslfqo8imj", member_id: "94", seq_head: "1442",…}
client_ops: [{optype: "InsertText", memberid: "94", timestamp: 1440522980677, position: 10, text: "3",…}]
es_id: "xxanna4crahd98q5gec8kslfqo8imj"
member_id: "94"
seq_head: "1442"
command: "sync_ops"

I wonder whether `member_id` is non-zero here.

**Comment by interele** [comment]
Sorry, I don't understand. I don't have a /apps/documents/ajax directory

**Comment by VicDeo** [comment]
@interele
> I don't have a /apps/documents/ajax directory

  true. 
There are sync requests that are sent to URL ending with `/apps/documents/ajax/otpoll.php`  after document is opened.
You need to open dev tools in browser (F12) and copy the data  sent to the server:

![chrome](https://cloud.githubusercontent.com/assets/991300/9493120/1cdf6e5c-4c06-11e5-9478-91b7e37facd5.png)

**Comment by interele** [comment]
![snapshot1](https://cloud.githubusercontent.com/assets/13849730/9493351/c383f83c-4bf6-11e5-981c-2a411e86e878.png)
 Is this what you want

**Comment by VicDeo** [comment]
@interele exactly. So you have zero memberId from the very beginning. What DB engine do you use?

**Comment by interele** [comment]
![snapshot1](https://cloud.githubusercontent.com/assets/13849730/9493407/202ddcec-4bf7-11e5-9b8a-39ad85080392.png)
This is from an account that works perfectly
All I did was log out of account that didn't work and log back into one that does
MariaDB

**Comment by VicDeo** [comment]
@interele is there any difference between these two accounts? Different user backends or something similar?

**Comment by VicDeo** [comment]
can you also check memberId in the response to the request `apps/documents/ajax/session/joinasuser/` when the document is opened
{"es_id":"zx0eo81baqq9m8yl4e27jjebgolqkm","genesis_url":"\/documents\/1f8de65f8383b5e83a4962805ae8ae18209a349a.odt","genesis_hash":"1f8de65f8383b5e83a4962805ae8ae18209a349a","file_id":"67","owner":"D","member_id":"95","title":"New Document.odt","permissions":27,"status":"success"}

at the  dev tools, 'Response' tab

**Comment by interele** [comment]
The only difference is that the one that works is me and I am a member of the group ITSupport
and the other one is a student they are members of Students and Intake2015. I have however
tried my colleagues account who is also in the same groups as me and hers doesn't work.
Mine didn't work for a while and after I disabled Documents and reenabled it is magically started
to work fine. Will give you the other info asap

**Comment by interele** [comment]
![snapshot1](https://cloud.githubusercontent.com/assets/13849730/9496107/a8fb181e-4c06-11e5-8d01-444486916741.png)

This is from a new document as the original document seems to be munged. If I try and open it in Documents now I get

![snapshot1](https://cloud.githubusercontent.com/assets/13849730/9496164/f346a06e-4c06-11e5-8532-61bd2e490ee4.png)
 
But the original ( unedited one ) is OK in Files

**Comment by interele** [comment]
"es_id":"cudtnpd1ihvziicl20052h7wx6y0hy","genesis_url":"\/documents\/5fc55a1f23a1183afaed4d90b50a8a69b3d69719.odt","genesis_hash":"5fc55a1f23a1183afaed4d90b50a8a69b3d69719","file_id":"28867","owner":"328edf36-af77-1034-9fbb-07ff2bc688ba","member_id":"0","title":"New Document.odt","permissions":27,"status":"success"}

**Comment by VicDeo** [comment]
@interele Are you able to execute queries on database? 
I need output for a query
`SELECT * from oc_documents_member WHERE es_id='cudtnpd1ihvziicl20052h7wx6y0hy'`

**Comment by VicDeo** [comment]
@interele @budachst Can you apply changes from https://github.com/owncloud/documents/pull/539 and retest?

**Comment by budachst** [comment]
Will do and report back.

**Comment by interele** [comment]
It doesn't work.

The first time you edit a document it works perfectly then Documents hangs

Scenario 

File as viewed in Files = ABC
File as viewed in Documents ABC123

Apply fix - restart

View file in Files = ABC
Edit file in Documents =ABC123 -> ABC123XYZ
View files in Files = ABC123XYZ

Seems to work ! Now go back and try and open file in Documents again
Documents just has the 'circular loading swirly clock' forever

It is now more consistent - previously my account worked and most of the others didn't
now all accounts exhibit the same hang ( mine doesn't work properly either )  Documents 
never seems to work again after working perfectly the first time. If you go into Documents
and add new document you just get the swirly thing. If you exit the page it has created the 
document

**Comment by interele** [comment]
BTW did query as requested - Empty Set
Doing SELECT * from oc_documents_member produces about 100 rows

**Comment by budachst** [comment]
I have tested some more today and it is still working for me. I actually conencted from two owncloud Accounts and edited the very same document, which saved and updated without any issue.

@interele I actually cleaned up the oc_documents_* tables prior to apply the patches. Maybe there're still old sessions saved, which shouldn't exist anymore.

**Comment by VicDeo** [comment]
@interele is memeber_id  still 0 for you?

**Comment by interele** [comment]
I think I've done something nasty to Documents - it doesn't work at all now
How can I re install it from scratch ?

**Comment by interele** [comment]
OK, got it all going again with the fix and so far so good.

**Comment by interele** [comment]
Works with all our test accounts

Thanks

M

**Comment by VicDeo** [comment]
Original report seems to be duplicate of https://github.com/owncloud/documents/issues/447
Everything else is  fixed in v0.10.2
closing
